lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py

```diff
@@ -217,238 +273,89 @@ def test_clear_tools_test70(monkeypatch, agent_config, pbench_run, pbench_cfg):
-        comma"""
-    groups_shouldnt_exists = []
-    for group in groups.split(","):
-        another_group = pbench_run / f"tools-v1-{group}"
-        another_group.mkdir(parents=True)
-        groups_shouldnt_exists.append(another_group)
-        for remote in remotes:
-            remote_host = another_group / f"{remote}.example.com"
-            remote_host.mkdir(parents=True, exist_ok=True)
-            pidstat7_tool = remote_host / "pidstat"
-            pidstat7_tool.touch()
-            mpstat7_tool = remote_host / "mpstat"
-            mpstat7_tool.touch()
-
-    command = ["pbench-clear-tools", f"--group={groups}", "--name=pidstat,mpstat"]
-    out, err, exitcode = pytest.helpers.capture(command)
-    assert b"" == out
-    for group in groups_shouldnt_exists:
-        assert group.exists() is False
-
-
-@pytest.mark.parametrize(
-    "groups", ["default", "default,custom1", "default,custom1,custom2"]
-)
 @pytest.mark.parametrize(
-    "remotes",
-    ["remote1", "remote1,remote2", "remote1,remote2,remote3", "remote2,remote4"],
+    "tools", ["pidstat,mpstat,iostat", "mpstat,vmstat"],
 )
-def test_clear_tools_test76(
-    monkeypatch, agent_config, pbench_run, pbench_cfg, groups, remotes
-):
-    """ Attempt to clear groups, by name, registered on multiple
-    hosts"""
-    tools_existence = []
-    for group in groups.split(","):
-        another_group = pbench_run / f"tools-v1-{group}"
-        another_group.mkdir(parents=True)
-
-        for remote in remotes.split(","):
-            fubar7_1_host = another_group / f"{remote}.example.com"
-            fubar7_1_host.mkdir(parents=True, exist_ok=True)
-            pidstat7_1_tool = fubar7_1_host / "pidstat"
-            pidstat7_1_tool.touch()
-            tools_existence.append(pidstat7_1_tool)
-
+def test_clear_tools_test86(tools_configuration, groups, remotes, tools):
+    """ Attempt to clear tools at various group-remote-tool locations where the
+    combination may not present on the hosts"""
     command = [
         "pbench-clear-tools",
-        f"--group={groups}",
-        "--remote=remote2.example.com,remote3.example.com",
+        f"--groups={groups}" if groups else "",
+        f"--remotes={remotes}" if remotes else "",
+        f"--name={tools}" if tools else "",
     ]
-    out, err, exitcode = pytest.helpers.capture(command)
+    command_refined = [x for x in command if x]
+
+    out, err, exitcode = pytest.helpers.capture(command_refined)
     assert b"" == out
-    for tool in tools_existence:
-        if any(x in str(tool) for x in ["remote2.example.com", "remote3.example.com"]):
-            assert tool.exists() is False
-        else:
-            assert tool.exists() is True
+    assert any(
+        f'pbench-clear-tools: invalid --group option "{x}"'.encode() in err
+        for x in groups.split(",")
+    )
+    assert any(f'No remote host "{x}"'.encode() in err for x in remotes.split(","))
```

Comment:
Rather than specifying nothing but bad options and then checking to see that all options show up in the errors, it would be better to specify one (or two) bad options, in addition to good options, and make sure that only the bad options show up in the errors.  And, it would be appropriate to check that the presence of the bad options does not prevent the good ones from being properly handled.

Reply:
The parameterization has a combination of 1. valid and a wrong group 2. all wrong groups, so the assertion should work with `any`